crates/nym-gateway-directory/src/entries/described_gateway.rs

```diff
@@ -23,6 +27,36 @@ impl DescribedGatewayWithLocation {
             .is_some()
     }
 
+    pub fn has_current_api_version(&self) -> bool {
+        self.has_v6_exit_build_timestamp() && self.has_current_build_version()
+    }
+
+    fn has_v6_exit_build_timestamp(&self) -> bool {
+        self.gateway
+            .self_described
+            .as_ref()
+            .map(|d| {
```

Comment:
Suggestion: 
.filter_map for parsing step
Then
.filter for date selection 
To keep it flat and non-nested 

Also you can split those two step into two functions

fn build_timestamp -> timestamp 

fn is_current_build -> bool